data-prepper-expression/src/test/java/org/opensearch/dataprepper/expression/ParseTreeCoercionServiceTest.java

```diff
@@ -97,36 +96,43 @@ void testCoerceTerminalNodeBooleanType() {
         assertThat(result, equalTo(testBoolean));
     }
 
-    @Test
-    void testCoerceTerminalNodeJsonPointerType() {
+    @ParameterizedTest
+    @MethodSource("provideSupportedJsonPointerValues")
+    void testCoerceTerminalNodeJsonPointerTypeSupportedValues(final Object testValue) {
         final String testKey1 = "key1";
         final String testKey2 = "key2";
-        final String testValue = "value";
         final String testJsonPointerKey = String.format("/%s/%s", testKey1, testKey2);
-        final Event testEvent = createTestEvent(Map.of(testKey1, Map.of(testKey2, testValue)));
+        final Event testEvent = testValue == null ? createTestEvent(new HashMap<>()) :
+                createTestEvent(Map.of(testKey1, Map.of(testKey2, testValue)));
         when(token.getType()).thenReturn(DataPrepperExpressionParser.JsonPointer);
         when(terminalNode.getSymbol()).thenReturn(token);
         when(terminalNode.getText()).thenReturn(testJsonPointerKey);
         final Object result = objectUnderTest.coercePrimaryTerminalNode(terminalNode, testEvent);
-        assertThat(result, instanceOf(String.class));
-        assertThat(result, equalTo(testValue));
+        if (testValue instanceof Double) {
+            assertThat(result, instanceOf(Float.class));
+            assertThat(result, equalTo(((Double) testValue).floatValue()));
+        } else {
+            assertThat(result, equalTo(testValue));
+        }
     }
 
-    @Test
-    void testCoerceTerminalNodeJsonPointerTypeMissingKey() {
-        final String testMissingKey = "missingKey";
-        final String testJsonPointerKey = "/" + testMissingKey;
-        final Event testEvent = createTestEvent(new HashMap<>());
+    @ParameterizedTest
+    @MethodSource("provideUnSupportedJsonPointerValues")
+    void testCoerceTerminalNodeJsonPointerTypeUnSupportedValues(final Object testValue) {
```

Comment:
Could `testValue` type be changed to Event and create the events in the `provideUnSupportedJsonPointerValues` method? I think it would make the test more readable.

Reply:
Actually the same method is used for both jsonpointer and escape jsonpointer test case in which keys are separate. So providing the value only will probably make it more compact and focused on the difference (testValue). Thought?